staking/integration-tests/tests/stability.rs

```diff
@@ -0,0 +1,439 @@
+                delegate,
+                slash,
+                undelegate,
+            },
+        },
+        publisher_caps::helper_functions::post_publisher_caps,
+        setup::{
+            setup,
+            SetupProps,
+            SetupResult,
+        },
+        solana::{
+            instructions::create_token_account,
+            utils::{
+                fetch_account_data,
+                fetch_account_data_bytemuck,
+                fetch_positions_account,
+            },
+        },
+        staking::{
+            helper_functions::initialize_new_stake_account,
+            pda::get_stake_account_custody_address,
+        },
+        utils::clock::{
+            advance_n_epochs,
+            get_current_epoch,
+        },
+    },
+    integrity_pool::{
+        state::pool::PoolData,
+        utils::{
+            clock::UNLOCKING_DURATION,
+            constants::MAX_PUBLISHERS,
+            types::FRAC_64_MULTIPLIER,
+        },
+    },
+    litesvm::LiteSVM,
+    quickcheck::{
+        Arbitrary,
+        Gen,
+        QuickCheck,
+    },
+    solana_sdk::{
+        pubkey::Pubkey,
+        signature::Keypair,
+        signer::Signer,
+    },
+    staking::{
+        error::ErrorCode as StakingErrorCode,
+        state::positions::{
+            Position,
+            PositionState,
+            Target,
+            TargetWithParameters,
+        },
+    },
+    std::{
+        cmp::min,
+        convert::TryInto,
+        fmt::Debug,
+    },
+};
+
+// These constants define the parameters of the stability test.
+const MAX_DELEGATION_AMOUNT: u64 = 2 * FRAC_64_MULTIPLIER;
+const MAX_UNDELEGATION_AMOUNT: u64 = FRAC_64_MULTIPLIER / 10;
+const NUM_DELEGATORS: u64 = 1_000;
+const NUM_OPERATIONS: u64 = 10_000;
+const MAX_CAP: u64 = 100 * FRAC_64_MULTIPLIER;
+
+#[derive(Debug)]
+struct KeypairWrapper(Keypair);
+
+impl Clone for KeypairWrapper {
+    fn clone(&self) -> Self {
+        KeypairWrapper(self.0.insecure_clone())
+    }
+}
+
+#[derive(Debug, Clone)]
+enum Operation {
+    Delegate {
+        publisher: usize,
+        delegator: usize,
+        amount:    u64,
+    },
+    Undelegate {
+        publisher: usize,
+        delegator: usize,
+        amount:    u64,
+    },
+    AdvanceDelegationRecord {
+        publisher: usize,
+        delegator: usize,
+    },
+    Advance {
+        cap: u64,
+    },
+    Slash {
+        publisher: usize,
+        amount:    u64,
+    },
+}
+
+#[derive(Clone)]
+struct StabilityTestProps {
+    operations: Vec<Operation>,
+    publishers: Vec<KeypairWrapper>,
+    delegators: Vec<KeypairWrapper>,
+}
+
+impl Debug for StabilityTestProps {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("StabilityTestProps")
+    }
+}
+
+impl Arbitrary for StabilityTestProps {
+    fn arbitrary(g: &mut Gen) -> Self {
+        let num_publishers = usize::arbitrary(g) % (MAX_PUBLISHERS - 10) + 10;
+
+        // TODO: make stable
```

Comment:
what does this comment mean?

Reply:
I wanted to make the Keypair generation stable, meaning it would generate the same keypairs everytime, but i think it doesn't matter as it will not effect the functionality in anyway.

Reply:
Ah I would've used the word `deterministic`